nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/processors/script/InvokeScriptedProcessor.java

```diff
@@ -0,0 +1,606 @@
+            }
+        } else if (instance != null) {
+            // If the script provides a Processor, call its onPropertyModified() method
+            try {
+                instance.onPropertyModified(descriptor, oldValue, newValue);
+            } catch (final Exception e) {
+                final String message = "Unable to invoke onPropertyModified from script Processor: " + e;
+                logger.error(message, e);
+            }
+        }
+    }
+
+    /**
+     * Reloads the script located at the given path
+     *
+     * @param scriptPath the path to the script file to be loaded
+     * @return true if the script was loaded successfully; false otherwise
+     */
+    private boolean reloadScriptFile(final String scriptPath) {
+        final Collection<ValidationResult> results = new HashSet<>();
+
+        try (final FileInputStream scriptStream = new FileInputStream(scriptPath)) {
+            return reloadScript(IOUtils.toString(scriptStream));
+
+        } catch (final Exception e) {
+            final ProcessorLog logger = getLogger();
+            final String message = "Unable to load script: " + e;
+
+            // If the module path has not yet been set, then this script is likely being loaded too early and depends
+            // on modules the processor does not yet know about. If this is the case, it will be reloaded later on
+            // property change (modules) or when scheduled
+            if (modulePath != null) {
+                logger.error(message, e);
+                results.add(new ValidationResult.Builder()
+                        .subject("ScriptValidation")
+                        .valid(false)
+                        .explanation("Unable to load script due to " + e)
+                        .input(scriptPath)
+                        .build());
+            }
+        }
+
+        // store the updated validation results
+        validationResults.set(results);
+
+        // return whether there was any issues loading the configured script
+        return results.isEmpty();
+    }
+
+    /**
+     * Reloads the script defined by the given string
+     *
+     * @param scriptBody the contents of the script to be loaded
+     * @return true if the script was loaded successfully; false otherwise
+     */
+    private boolean reloadScriptBody(final String scriptBody) {
+        final Collection<ValidationResult> results = new HashSet<>();
+        try {
+            return reloadScript(scriptBody);
+
+        } catch (final Exception e) {
+            final ProcessorLog logger = getLogger();
+            final String message = "Unable to load script: " + e;
+
+            // If the module path has not yet been set, then this script is likely being loaded too early and depends
+            // on modules the processor does not yet know about. If this is the case, it will be reloaded later on
+            // property change (modules) or when scheduled
+            if (modulePath != null) {
```

Comment:
seems like something a bit wonky is happening here.  So if one enters a valid script then enters a buggy script the original value is set and not being replaced.  Then because if there is an invalid script and no modulepath the error is being buried.  Definitely need the old script/processor that was compiled to get wiped out and we need the errors to always make it to the log/bulletin.